server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -553,11 +552,23 @@ private void setTranslogFlushThresholdSize(ByteSizeValue byteSizeValue) {
     }
 
     private void setTranslogRetentionSize(ByteSizeValue byteSizeValue) {
```

Comment:
do we need to call this setter in the `IndexSettings` constructor? Otherwise this check is bypassed on initial set.

Reply:
Yes, we already this in the constructor https://github.com/elastic/elasticsearch/pull/45473/files#diff-af854c6d1b20d51b677cd89fd2ed5e07R493.